### PR TITLE
[DF] Treat RVec<bool> as a collection in Display 

### PR DESCRIPTION
**Pull request by eguiraud**
This fixes the report in ROOT-10889.

### CODE REVIEW

**Comment by eguiraud** [comment]
@stwunsch does this resolve the problem reported in ROOT-10889 ?

File: tree/dataframe/inc/ROOT/RDF/RDisplay.hxx [hunk 1]
```diff
@@ -154,6 +153,26 @@ private:
       return false; // do not treat this as a collection
    }
 
+   ////////////////////////////////////////////////////////////////////////////
+   /// AddInterpreterString overload for arrays of booleans.
+   ///
+   /// \param[in] boolArr The character array to convert to string representation
```
Comment (stwunsch) [inline tree/dataframe/inc/ROOT/RDF/RDisplay.hxx:159]:
```suggestion
   /// \param[in] boolArr The bool array to convert to string representation
```

File: tree/dataframe/inc/ROOT/RDF/Utils.hxx [hunk 2]
```diff
@@ -54,7 +54,10 @@ using namespace ROOT::RDF;
 
 /// Check for container traits.
 ///
-/// Note that we don't recognize std::string as a container.
+/// Note that for all uses in RDF we don't want to classify std::string as a container.
+/// Template specializations of IsDataContainer make it return `true` for std::span<T>, std::vector<bool> and
+/// RVec<bool>, which we do want to count as containers even though they don not satisty all the traits tested by the
```
Comment (stwunsch) [inline tree/dataframe/inc/ROOT/RDF/Utils.hxx:59]:
```suggestion
/// RVec<bool>, which we do want to count as containers even though they do not satisfy all the traits tested by the
```

File: tree/dataframe/inc/ROOT/RDF/RDisplay.hxx [hunk 3]
```diff
@@ -154,6 +153,26 @@ private:
...
+   /// This specialization for arrays of booleans skips the cling::printValue
+   /// (i.e. appends nothing to the stream) and directly writes to fCollectionsRepresentations the
+   /// string representation of the array of chars.
+   bool AddInterpreterString(std::stringstream &, ROOT::RVec<bool> &boolArr, const int &index)
+   {
+      fCollectionsRepresentations[index].reserve(boolArr.size());
+      for (bool b : boolArr)
+         fCollectionsRepresentations[index].push_back(b ? "true" : "false");
+
+      return true; // treat this as a collection
```
Comment (stwunsch) [inline tree/dataframe/inc/ROOT/RDF/RDisplay.hxx:172]: I think that comment got lost in my previous review:

The line
> return true; // treat this as a collection
contradicts what is stated in the doxygen markup
> \return true, the event is not a collection

**Comment by stwunsch** [comment]
> @stwunsch does this resolve the problem reported in ROOT-10889 ?

Yes, definitely! All other issues found there are due to my misunderstanding of `Print` vs `AsString`.

File: tree/dataframe/inc/ROOT/RDF/RDisplay.hxx (hunk 3 above)
Comment (eguiraud) [reply tree/dataframe/inc/ROOT/RDF/RDisplay.hxx:172]: ah thanks, copy-paste mistake

File: tree/dataframe/inc/ROOT/RDF/RDisplay.hxx [hunk 4]
```diff
@@ -154,6 +153,26 @@ private:
       return false; // do not treat this as a collection
    }
 
+   ////////////////////////////////////////////////////////////////////////////
+   /// AddInterpreterString overload for arrays of booleans.
+   ///
+   /// \param[in] boolArr The character array to convert to string representation
+   /// \param[in] index To which column the event belongs
+   /// \return true, the event is not a collection
```
Comment (eguiraud) [inline tree/dataframe/inc/ROOT/RDF/RDisplay.hxx:161]:
```suggestion
   /// \return true, the event is a collection
```

File: tree/dataframe/inc/ROOT/RDF/RDisplay.hxx (hunk 3 above)
Comment (eguiraud) [reply tree/dataframe/inc/ROOT/RDF/RDisplay.hxx:172]: fixed

**Comment by eguiraud** [comment]
> We have tests for IsDataContainer in tree/dataframe/test/dataframe_utils.cxx, you could add an additional test to cover RVec<bool>

done!

**Pull request merged**